parser/parser.go

```diff
@@ -85,19 +92,35 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 				Result: FAIL,
 				Output: make([]string, 0),
 			})
-		} else if matches := regexResult.FindStringSubmatch(line); len(matches) == 5 {
-			if matches[4] != "" {
-				coveragePct = matches[4]
+
+			// clear the current build package, so output lines won't be added to that build
+			capturedPackage = ""
+		} else if matches := regexResult.FindStringSubmatch(line); len(matches) == 6 {
+			if matches[5] != "" {
+				coveragePct = matches[5]
 			}
 
 			// all tests in this package are finished
-			report.Packages = append(report.Packages, Package{
+			pack := Package{
 				Name:        matches[2],
+				CoveragePct: coveragePct,
 				Time:        parseTime(matches[3]),
 				Tests:       tests,
-				CoveragePct: coveragePct,
-			})
+			}
+
+			if matches[4] == "[build failed]" {
+				// the build of the package failed, inject a dummy test into the package
+				// which indicate about the failure and contain the failure description.
+				pack.Tests = []*Test{
```

Comment:
This works, but I'd prefer if you just add a new `Test` to the `tests` slice to keep it similar with what we do on line 90 for example.